extensions/amp-date-picker/1.0/component/base-date-picker.tsx

```diff
@@ -0,0 +1,53 @@
+import objStr from 'obj-str';
+import {
+  DayPicker,
+  DayPickerRangeProps,
+  DayPickerSingleProps,
+} from 'react-day-picker';
+
+import * as Preact from '#preact';
+
+import {DayButton} from './day-button';
+import {useDatePickerContext} from './use-date-picker-context';
+
+import {useStyles} from '../component.jss';
+
+type BaseDatePickerProps = (DayPickerSingleProps | DayPickerRangeProps) & {
+  isOpen: boolean;
+};
+
+export function BaseDatePicker({isOpen, ...props}: BaseDatePickerProps) {
+  const classes = useStyles();
+  const {
+    formatMonth,
+    formatWeekday,
+    isDisabled,
+    locale,
+    numberOfMonths,
+    today,
+  } = useDatePickerContext();
+
+  return (
+    <div
+      aria-label="Calendar"
+      hidden={!isOpen}
+      class={objStr({
+        [classes.unmounted]: !isOpen,
+        [classes.dayPicker]: true,
+      })}
+    >
+      <DayPicker
+        components={{Day: DayButton}}
+        disabled={[isDisabled]}
+        formatters={{
+          formatCaption: formatMonth,
+          formatWeekdayName: formatWeekday,
+        }}
```

Comment:
Something for us to maybe worry about later if performance becomes a slight issue would be to memoize these objects between renders. But I feel like that's an optimization we can make later, if we need it.